blog/content/post/2020-11-30-coreboot_for_power.md

```diff
@@ -0,0 +1,154 @@
+tasks include initialisation of the rest of the hardware, e.g. PCI Express host
+bus controllers. It also implements OPAL ([OpenPOWER Abstraction Layer](https://open-power.github.io/skiboot/doc/opal-spec.html#what-is-opal)),
+which is an interface to access firmware services from an OS. This is similar to
+the UEFI runtime services or legacy BIOS interrupt calls.
+
+Next link in this chain is [Skiroot](https://github.com/open-power/linux) (a
+small set of patches on top of the mainstream Linux kernel) accompanied by
+[Petitboot](https://git.raptorcs.com/git/talos-petitboot/) (userspace
+application). The latter is a bootloader, it uses kexec to load the final
+operating system from any device that can be mounted by Linux, including network
+boot.
+
+There are also other pieces of firmware running on different chips. Those are
+responsible for thermal and power management or CAPI (coherent accelerator
+processor interface) support. Most of them are loaded by the Hostboot.
+
+A list of links to more detailed descriptions of all of the mentioned components
+can be found [here](https://wiki.raptorcs.com/wiki/OpenPOWER_Firmware).
+
+## coreboot's place in the OpenPOWER boot chain
+
+OpenPOWER separates individual tasks required in the boot process in a slightly
+different manner than coreboot. In coreboot, a great emphasis is placed on the
+separation of one-time initialisation (coreboot stages) and runtime services
+(payload). Skiboot mixes those two - it does the hardware initialisation usually
+performed by a ramstage in coreboot, but it also installs OPAL services, which
+is better suited for a payload in a coreboot's world.
+
+With the above in mind we decided that it would be best to start with porting
+just the Hostboot part, at least for the time being. Earlier stages are run by a
+separate hardware, and dividing Skiboot into "hardware" and "services" parts at
+this stage may introduce additional bugs. It is safer to implement one component
+at a time than fail trying to do them all in one go.
+
+## Benefits
+
+We can define two classes of possible benefits: one is what coreboot and its
+community can gain, the other one is what end users get from it.
+
+#### For users
+
+Boot time reduction was one of the main reasons for doing this port. Hostboot
+itself takes more than 1 minute to run. This is mostly spent on accessing flash
+memory. It is split into multiple PNOR (Processor NOR, POWER name for flash)
+partitions, summing up to just below 32 megabytes. It runs from cache until it
+trains main memory, so it is limited to just 10 MB (size of L3 cache) split
+between two cores. To make it possible to load and execute that amount of code,
+[Hostboot uses on-demand paging](https://youtu.be/fTLsS_QZ8us?t=1559). It is
+similar to swap used by operating systems, except that it uses ROM instead of
+writable media, so only code and constant data can be discarded from memory (L3
+cache in this case).
```

Comment:
This isn't true.  We have writable sections of the pnor, but they are only used to store a few writable persistent attributes and the VPD caches.